docs/source/class_basics.rst

```diff
@@ -151,6 +151,112 @@ concrete. As with normal overrides, a dynamically typed method can
 implement a statically typed abstract method defined in an abstract
 base class.
 
+.. _metaclasses:
```

Comment:
I think ``class_basics`` is a wrong section for this. Metaclasses are not really basic (although they definitely related to classes). I propose to put this in ``kinds_of_types``.

Reply:
I agree that it is not class basics, but it's not some kind of type either. In my opinion "class basics" should be the first subsection in the section, which should be named something like "Classes and Protocols", or "Classes, Metaclasses and Protocols" or some other generic name. Or maybe just split this section to three.

Reply:
> but it's not some kind of type either

Why not? Everything that is allowed in type context is a type, and we have a special section there about ``Type[...]`` that seems related. (It may be even instructive to explain what is the difference between ``Type[A]`` and ``M`` for ``class A(metaclass=M): ...``)

Reply:
Because metaclass is a runtime thing, which is quite different from most of the stuff in that page. The difference is indeed important, but that explanation can be given independently.

Reply:
> Because metaclass is a runtime thing, which is quite different from most of the stuff in that page.

Most things there are different from other things to a similar extent, this is normal. I would also split this discussion and put the "Gotchas and limitations of metaclass support" in common_issues.